src/gen/conditions.rs

```diff
@@ -233,7 +273,7 @@ pub struct SpendConditionSummary {
     pub seconds_relative: u64,
     pub seconds_absolute: u64,
     // all create coins. Duplicates are consensus failures
-    pub create_coin: HashSet<(Vec<u8>, u64)>,
+    pub create_coin: HashSet<NewCoin>,
```

Comment:
a plain tuple isn't sufficient anymore. The hint cannot be part of what's hashed for this `HashSet`. The point of putting the coins in a hash set is to ensure there are not duplicate outputs (i.e. identical puzzle hash and identical amount). The hint doesn't contribute to make a coin unique, so it has to be ignored for this purpose.